pycode/memilio-simulation/memilio/simulation/pybind_util.h

```diff
@@ -52,7 +96,34 @@ decltype(auto) pybind_pickle_class(pybind11::module_& m, const char* name)
                 throw std::runtime_error(object.error().formatted_message());
             }
         }));
-    return pickle_class;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, NoPicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    return cls;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, TryPicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    // Bind the class depending on its features
+    if constexpr (is_serializable<mio::PickleSerializer, T>::value) {
+        pybind_pickle_class<T, Args...>(cls);
+    }
+    return cls;
+}
+
+template <class T, class... Args>
+decltype(auto) _bind_class(pybind11::module& m, std::string const& name, ForcePicklingTag) {
+    decltype(auto) cls = pybind11::class_<T, Args...>(m, name.c_str());
+    pybind_pickle_class<T, Args...>(cls);
+    return cls;
+}
+
+template <PickleFlag F, class T, class... Args>
+decltype(auto) bind_class(pybind11::module& m, std::string const& name) {
+    return _bind_class<T, Args...>(m, name, typename PicklingTag<F>::type{});
 }
```

Comment:
My thinking was to not divide the main class T and base classes Args. But i see that the class T is the most important part.

Reply:
Good point. I would still prefer pickle<Derived, PicklingTag, Base> over the alternative. I think that's readable enough. 

I guess the most generic version would be to make the PicklingTag a part of `Args...` (can't be an enum then) and iterate over Args at compile time to find it. Then it could be anywhere, and other flags could be added in the same way when needed. But I think that's not worth the effort now.